_posts/2020-04-03-what-does-appsync-do.md

```diff
@@ -0,0 +1,200 @@
+- RELATIONAL_DATABASE
+
+In this case, we will use DynamoDb as our data source.
+
+```yml
+CoffeeShopDynamoDB:
+  Type: AWS::DynamoDB::Table
+  Properties:
+    TableName: "CoffeeShopTable"
+    AttributeDefinitions:
+      - AttributeName: "id"
+        AttributeType: "S"
+      - AttributeName: "resourceId"
+        AttributeType: "S"
+    KeySchema:
+      - AttributeName: "id"
+        KeyType: "HASH"
+      - AttributeName: "resourceId"
+        KeyType: "RANGE"
+    ProvisionedThroughput:
+      ReadCapacityUnits: 3
+      WriteCapacityUnits: 1
+    Projection:
+      ProjectionType: "ALL"
+
+CoffeeShopDynamoDBDataSource:
+  Type: AWS::AppSync::DataSource
+  Properties:
+    ApiId: !GetAtt CoffeeShopAppSyncApi.ApiId
+    Name: CoffeeShopDynamoDBDataSource
+    Description: Contain order data from coffee shop
+    Type: “AMAZON_DYNAMODB”
+    ServiceRoleArn: arn:aws:iam::xxxxxxxxxxxx:role/xxxxxxxxx
+    DynamoDBConfig:
+      TableName: CoffeeShopTable
+      AwsRegion: ap-southeash-2
+```
+
+From the template, we are provisioning new dynamoDB table for the API. It has the attribute id as the hash key and resourceId as the range key for the table.
+
+We are connecting the AppSync data source with the newly created dynamoDB table by specifying the table name. You can connect to an existing data source if you already have one.
+
+### Step 4: Create the Resolver
+
+Adding resolver for a mutation, query or subscription automatically coverts arguments that match the schema into database operation.
+
+```yml
+CreateOrderResolver:
+  Type: AWS:: AppSync::Resolver
+  Properties:
+    ApiId: !GetAtt CoffeeShopAppSyncApi.ApiId
+    TypeName: Mutation
+    FieldName: createOrder
+    DataSourceName: !GetAtt CoffeeShopDynamoDBDataSource.Name
+    RequestMappingTemplate: |
+      {
+        "version" : "2017-02-28",
+        "operation" : "PutItem",
+        "key" : {
+          "id": { "S" : "${ctx.args.orderInput.id}" },
+          "resourceId": { "S": "COFFEE" }
+        },
+      }
+    ResponseMappingTemplate: |
+        $util.toJson($context.result)
+
+GetOrderResolver:
+  Type: AWS:: AppSync::Resolver
+  Properties:
+    ApiId: !GetAtt CoffeeShopAppSyncApi.ApiId
+    TypeName: Query
+    FieldName: getOrder
+    DataSourceName: !GetAtt CoffeeShopDynamoDBDataSource.Name
+    RequestMappingTemplate: |
+      {
+          "version": "2017-02-28",
+          "operation": "GetItem",
+          "key": {
+            "id": { "S": "${ctx.args.id}" },
+            "resourceId": { "S": "COFFEE" }
+          }
+        }
+    ResponseMappingTemplate: |
+        $util.toJson($context.result)
+```
+
+# Conclusion
+
+We had a brief introduction on how you can provision AWS AppSync with DynamoDb as the data source by using CloudFormation.
+
+Overall I think that AWS AppSync is a good starting point for developers who want to start building an API for their application. This is best suited for applications where real-time data and high speeds are crucial.
```

Comment:
"...and high speeds are crucial." -> "...and development velocity is crucial"

I'm not sure if requesting data from AppSync is any faster/slower than going through API Gateway.